engine/src/main/java/org/destinationsol/game/FarObject.java

```diff
@@ -16,19 +16,85 @@
+ * the player/camera gets close to them again, at which point they are again transformed into their specific
+ * {@code SolObjects}. {@code FarObject}s generally have no body associated with them and do not inetract with the game much.
+ */
 public interface FarObject {
+
+    /**
+     * Denotes whether the object should be removed as of the time of calling.
+     * <p>
+     * This method should return false if the object is meant to persist for the whole game, like for instance
+     * {@link StarPort}. Otherwise, it usually checks for its {@link RemoveController} as to whether it should be removed.
+     *
+     * @param game Game this object belongs to.
+     * @return Boolean denoting whether the object should be removed.
+     */
     boolean shouldBeRemoved(SolGame game);
 
+    /**
+     * Creates a new {@link SolObject} similar to the one used for creation of this object by call to {@link SolObject#toFarObject()}
+     * <p>
+     * The created object should be by all terms similar or same as the {@code SolObject} that created this object.
+     *
+     * @param game Game this object belongs to.
+     * @return SolObject representation of this object.
+     */
     SolObject toObject(SolGame game);
 
+    /**
+     * Called on every game's frame, allows for handling of object-specific runtime needs.
+     * <p>
+     * This method should be much simpler than the {@link SolObject#update(SolGame)}. It can be used for instance for
+     * sort of simplified movement for ships.
+     *
+     * @param game Game this object belongs to.
+     */
     void update(SolGame game);
 
+    /**
+     * Returns the approximate radius the object is taking up, computed from the point returned by {@link #getPosition()}.
+     * <p>
+     * This radius can be used for easy checking of space taken by object, and should be taken in account when computing a distance.
+     *
+     * @return Approximate radius of this object.
+     */
     float getRadius();
 
+    /**
+     * Returns a position of the center of this object.
+     * <p>
+     * This can be considered static through the life of the object, unless it specifies a specific movement algorithm in
+     * {@link #update(SolGame)}. This method is used in combination with {@link #getRadius()} to compute distance to this object.
+     *
+     * @return Position of this object.
+     */
     Vector2 getPosition();
 
+    /**
+     * Used for retrieval of object's debug string.
+     * <p>
+     * Any kind of information can be used in the debug string, if you don't need/want to display any debug information,
+     * you can freely have this method return null. To display debug strings in-game, set the flag {@link DebugOptions#OBJ_INFO}.
+     * These strings should then be rendered in the proximity of their objects.
+     *
+     * @return Debug string with information about the object.
```

Comment:
Traditionally I've found param and return statements usually start with the data type, so in this case maybe "String with debug information about the object" but that's pretty trivial and just a hunch, I don't know if it is a convention.